test/e2e/autoscaling/dns_autoscaling.go

```diff
@@ -104,7 +105,8 @@ var _ = SIGDescribe("DNS horizontal autoscaling", func() {
 
 	// This test is separated because it is slow and need to run serially.
 	// Will take around 5 minutes to run on a 4 nodes cluster.
-	f.It(f.WithSerial(), f.WithSlow(), "kube-dns-autoscaler should scale kube-dns pods when cluster size changed", func(ctx context.Context) {
+	// TODO(upodroid) This test will be removed in 1.33 when kubeup is removed
```

Comment:
Is that correct? Wasn't the goal instead to get it modified so that it doesn't depend on KubeUp?

Reply:
For some tests, I did modify tests where it is possible and more importantly makes sense and I can run it on any kubernetes distro.

- https://github.com/kubernetes/kubernetes/pull/121007
- https://github.com/kubernetes/kubernetes/pull/121422

The test that I marked for "removal" in this PR does something like this:
- Change DNS autoscaling configmap (Good)
- Check pod count (Good)
- Increase the GCP Worker Node count (Bad)
- Check the pod count (Good)
- Set the node count back to 3 (Bad)

This is very problematic and relies on kubeup implementation and more importantly won't run on other kubernetes distros.

Reply:
What's the process for getting such tests fixed or removed? Is the test owner (in this case, SIG Autoscaling) being notified about the upcoming removal, if that is the intention?

Reply:
1. Add the KubeUp label to all the tests that should be removed
2. Send an email at the end of 1.30 dev cycle that the tests are being deleted when 1.33 cycle starts.

Reply:
That sounds fair to me.

Reply:
I think it depends on the test, for tests that are clearly giving us core coverage for Kubernetes we need to not depend on others to maintain the coverage bar unfortunately. For this case maybe not so much.

Reply:
> we need to not depend on others to maintain the coverage

Someone develops a test that is important for core coverage and then SIG Testing owns long term maintenance of it?

That's not my understanding of the charter, in particular considering https://github.com/kubernetes/community/blob/master/sig-testing/charter.md#out-of-scope:
> We are not responsible for writing, fixing nor actively troubleshooting tests for features or subprojects owned by other SIGs

There's one loophole: if we are making changes to infrastructure, it's on us to ensure that we don't loose coverage because of those changes, either by updating tests ourselves or working with other SIGs. Removing KubeUp probably falls into that  category.

Reply:
> There's one loophole: if we are making changes to infrastructure, it's on us to ensure that we don't loose coverage because of those changes, either by updating tests ourselves or working with other SIGs. Removing KubeUp probably falls into that category.

This.

Reply:
Also, charters aside: We don't want to cause regressions on core coverage at minimum, nor would we want anyone else's changes to cause this. So as a more general statement if we're putting in a big change to CI we may also have to continue to patch the tests (as Mahamed has been doing).

I don't know that _this_ test is a good example though, it's a specific addon. On the other hand, this is some _old_ work that has been working fine so it's not surprising that other people are not currently actively working on a test like this.

Reply:
Sounds like we are on the same page.

There's just one aspect open: should tests that get labeled because they cannot be fixed immediately still run until they finally get removed some releases later? I suspect that they don't run anymore at all when adding the `[Feature:KubeUp]`.

Reply:
This question brings us back to the purpose of this PR and how to achieve that purpose.

If they should run and my suspicion is correct, then labeling them with `f.WithLabel("KubeUp")` (i.e. without `Feature:` prefix) and adding an explicit skip for that label in the `pull-kubernetes-e2e-gce-canary` job may be the better solution.

Reply:
https://testgrid.k8s.io/google-gce#gce-cos-master-serial used to run "kube-dns-autoscaler should scale kube-dns pods when cluster size changed", now it doesn't.

Reply:
This PR has been reverted and `[KubeUp]` is now being used instead in https://github.com/kubernetes/kubernetes/pull/121880.

Reply:
+1, thank you! Sorry I've been only semi-available this week w/ post-kubecon cold 🤒